### PR TITLE
style: Move font-family outside of mako

### LINKED ISSUE #19355: Move font-family outside of mako.
I'll take this as a sub-issue of #19015 
Besides, this will be the last one of `font.mako.rs` 🎉

### PR DESCRIPTION
**Pull request by CYBAI**
This is a sub-PR of #19015 
Besides, this is the last PR for `font.mako.rs`! 🎉 
r? emilio 

---
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #19355 
- [x] These changes do not require tests

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/19366)

### CODE REVIEW

File: components/style/values/computed/font.rs [hunk 1]
```diff
@@ -215,6 +227,419 @@ impl ToAnimatedValue for FontSize {
...
+            syntax: FamilyNameSyntax::Identifiers(serialization),
+        }))
+    }
+
+    #[cfg(feature = "gecko")]
+    /// Return the generic ID for a given generic font name
+    pub fn generic(name: &Atom) -> (structs::FontFamilyType, u8) {
+        use gecko_bindings::structs::FontFamilyType;
+        if *name == atom!("serif") {
+            (FontFamilyType::eFamily_serif,
```
Comment (emilio) [inline components/style/values/computed/font.rs]: nit: Indentation looks weird here, I'd put all either on the same line, or:

```rust
(FontFamilyType::eFamily_serif,
 structs::kGenericFont_serif)
```

Though no big preference...

File: components/style/values/computed/font.rs [hunk 2]
```diff
@@ -215,6 +227,419 @@ impl ToAnimatedValue for FontSize {
...
+                SingleFontFamily::FamilyName(FamilyName {
+                    name: name.clone(),
+                    syntax: FamilyNameSyntax::Identifiers(serialization),
+                })
+            },
+            FontFamilyType::eFamily_named_quoted => SingleFontFamily::FamilyName(FamilyName {
+                name: (&*family.mName).into(),
+                syntax: FamilyNameSyntax::Quoted,
+            }),
+            x => panic!("Found unexpected font SingleFontFamily: {:?}", x),
```
Comment (emilio) [inline components/style/values/computed/font.rs]: nit: Found unexpected `FontFamilyType`, maybe?

File: components/style/values/computed/font.rs [hunk 3]
```diff
@@ -215,6 +227,419 @@ impl ToAnimatedValue for FontSize {
...
+                write!(dest, "{}", name)
+            },
+        }
+    }
+}
+
+#[cfg(feature = "servo")]
+#[derive(Clone, Debug, Eq, Hash, MallocSizeOf, PartialEq)]
+/// A list of SingleFontFamily
+pub struct FontFamilyList(Vec<SingleFontFamily>);
```
Comment (emilio) [inline components/style/values/computed/font.rs]: nit: This could be `Box<[SingleFontFamily]>`.

**Review by emilio — COMMENTED** [review]
Looks great, just minor nits.

Thanks a lot for knocking out `font.mako.rs`!!! \o/

File: components/style/values/computed/font.rs (hunk 1 above)
Comment (CYBAI) [reply components/style/values/computed/font.rs]: Oh, I didn't notice it. In the same line might be a little tedious?
Please let me make them in the same indentation in different lines.

File: components/style/values/computed/font.rs (hunk 2 above)
Comment (CYBAI) [reply components/style/values/computed/font.rs]: Yes, nice catch! Thanks!

**Pull request merged**